lib/shared-method.js

```diff
@@ -134,100 +151,90 @@ SharedMethod.fromFunction = function(fn, name, sharedClass, isStatic) {
     http: fn.http,
     rest: fn.rest
   });
-}
+};
+
+SharedMethod.prototype.getReturnArgDescByName = function(name) {
+  var returns = this.returns;
+  var desc;
+  for (var i = 0; i < returns.length; i++) {
+    desc = returns[i];
+    if (desc && ((desc.arg || desc.name) === name)) {
+      return desc;
+    }
+  }
+};
 
 /**
  * Execute the remote method using the given arg data.
  *
+ * @param {Object} scope `this` parameter for the invocation
  * @param {Object} args containing named argument data
- * @param {Function} fn callback `fn(err, result)` containing named result data
+ * @param {Object=} remotingOptions remote-objects options
+ * @param {Function} cb callback `fn(err, result)` containing named result data
  */
 
-SharedMethod.prototype.invoke = function (scope, args, fn) {
+SharedMethod.prototype.invoke = function(scope, args, remotingOptions, ctx, cb) {
   var accepts = this.accepts;
   var returns = this.returns;
   var errors = this.errors;
   var method = this.getFunction();
   var sharedMethod = this;
   var formattedArgs = [];
-  var result;
+
+  if (typeof ctx === 'function') {
+    cb = ctx;
+    ctx = undefined;
+  }
+
+  if (cb === undefined && typeof remotingOptions === 'function') {
+    cb = remotingOptions;
+    remotingOptions = {};
+  }
 
   // map the given arg data in order they are expected in
-  if(accepts) {
-    for(var i = 0; i < accepts.length; i++) {
+  if (accepts) {
+    var invalidArgs = [];
```

Comment:
This should be the `return` of `doesNotContain`, instead of changing a value out-of-scope